pyteal/ast/if_.py

```diff
@@ -63,4 +69,27 @@ def __str__(self):
     def type_of(self):
         return self.thenBranch.type_of()
 
+    def Then(self, thenBranch: Expr):
+        if self.elseBranch:
+            if not isinstance(self.elseBranch, If):
+                raise TealInputError("Else-Then block is malformed")
+            self.elseBranch.thenBranch = thenBranch
+        else:
+            self.thenBranch = thenBranch
+        return self
+
+    def ElseIf(self, cond):
```

Comment:
I wonder if multiple `ElseIf()`s should even be supported because existing `If()`s correlate to ternaries.

Reply:
Ideally it would be possible to create an arbitrary number of `ElseIf()`s, since the following are syntactically the same, with the second being easier to read in my opinion:

```python
# nested chain of if/elseifs
If(1)
  .Then(2)
  .Else(If(3)
    .Then(4)
    .Else(If(5)
      .Then(6)
      .Else(7)
    )
  )

# direct chained if/elseifs
If(1)
  .Then(2)
  .ElseIf(3)
  .Then(4)
  .ElseIf(5)
  .Then(6)
  .Else(7)
```

I have a suggestion to support an arbitrary number of `ElseIf()`s using recursion. During a call to `ElseIf(cond)`:
* If `elseBranch` is None, set `elseBranch` to `If(cond)`
* If `elseBranch` is not None and is an instance of `If`, call `elseBranch.ElseIf(cond)`
* If `elseBranch` is not None and is not an instance of `if`, raise an error (this means `ElseIf` was called after `Else`)

You would also need to modify `Then` and `Else` to recursively carry on the operation to the `elseBranch`, similar to this.

Reply:
I like your suggestion! I will try this out and write some more unit tests ...